src/CachePlugin.php

```diff
@@ -69,15 +69,42 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
         $cacheItem = $this->pool->getItem($key);
 
         if ($cacheItem->isHit()) {
-            // return cached response
             $data = $cacheItem->get();
-            $response = $data['response'];
-            $response = $response->withBody($this->streamFactory->createStream($data['body']));
+            if (isset($data['expiresAt']) && time() < $data['expiresAt']) {
```

Comment:
in my opinion, deploying a major version change could have the requirement to clear the cache.

as we add this now, i would assume this to be a non-issue, as cache arrays will be updated by the time we release version 2, unless somebody jumps from 1.0 to 2.0 directly.